Adding first pass meta avatar documentation.

stephen-radial opened the pull request:
First pass on setting up meta avatar user guide.

stephen-radial committed:
Adding first pass meta avatar documentation.

maxweisel reviewed (changes requested):
I can't seem to get this working. I think the image references aren't set up correctly.

You can test it by cloning this repo and running `npm install` and `npm run dev` to start the dev server.

Take a look at the other guides for how images are embedded vs videos. It's a slightly easier process. You'll also want to add a link to this guide in `sidebars.ts` in order to be able to access it.

stephen-radial committed:
Modifying image links, adding guide to side bar and adding the missing half of the guide...

stephen-radial commented:
Those are all great notes.  I was also testing mine in Visual Studio Code which, as it turns out, can hang onto things in memory and won't necessarily ask you to save, so the second half of the guide wasn't actually in there.

Tweaked the photo references, add the guide to the sidebar and added the missing info and formatting that hadn't been saved the first time.

Images will still need tweaking as part of the review they vary a bit in resolution (and I really don't like the un-even horizon on the player wave image.)

stephen-radial committed:
Adding a quick note/link on ensuring that the user's meta app is setup properly.

maxweisel commented on docs/guides/setting-up-a-networked-meta-avatar.md:
```diff
@@ -0,0 +1,103 @@
+
+## Setup your App for Meta Avatars
+
+Before your app can load avatar data from Meta's servers, it will need permission to do so.  Ensure that your app is [configured as required for Meta Avatar usage](https://developer.oculus.com/documentation/unity/meta-avatars-app-config/), and that you have set your Oculus Platform Settings to point at the right app IDs
```
The linked Meta documentation doesn't show how to get the app ID and configure Oculus Platform Settings for the project. I don't think it's enough to say "that you have set your Oculus Platform Settings to point at the right app IDs". Even as a seasoned developer, I don't know what that means unless I've done it before.

stephen-radial replied:
Added a bunch of documentation around navigating the Meta Developer website and setting up your app ids and permissions (and some notes for Normcore as well).

maxweisel commented on docs/guides/setting-up-a-networked-meta-avatar.md:
```diff
@@ -0,0 +1,103 @@
+## Setting up your VR environment
+
+Start by creating a blank unity VR scene, with the camera properly tracked to the player's head.
+
```
I feel like if someone is beginner enough to need to be told to set up a camera that tracks the player's head, we should also explain how to do that? Showing a screenshot with the TrackedPoseDriver isn't going to be immediately obvious to a beginner that you need to add that component in order to make the camera track the head imo.

OculusSetFloorTracking is really a component that exists for the example scenes so they don't need to depend on the Oculus plugin. However, in this case, it's not possible to use Meta Avatars without the Oculus plugins. I don't think we should be telling folks to use it this component. They should configure this through OVRManager when they set that up later on in this guide.

stephen-radial replied:
Swapped out the camera picture for a 2021 version which doesn't have the OculusSetFloorTracking.

maxweisel commented on docs/guides/setting-up-a-networked-meta-avatar.md:
```diff
@@ -0,0 +1,103 @@
+
+For the avatar system to initalize properly, your scene will require a number of components in it (**OvrAvatarManager**, **AvatarLODManager**, **GPUSkinningConfiguraiton** and **OvrAvatarShaderManager**).  The easiest way get these is to install the Meta Avatar SDK Sample Scenes, and use one of the prefabs they provide as part of that.
+
+First install the sample scenes through the package manager.  If you don't see the Meta Avatar SDK, you may need to [add the Meta Avatar SDK to your package list manually.](https://developer.oculus.com/downloads/package/meta-avatars-sdk/)
```
This link does not show me how to do add the meta avatars SDK to my package list manually?

maxweisel replied:
I'm now stuck at this point. I don't know how to get this package installed. This page has a download button which brings me to the asset store or to a UPM registry page. I don't want to go the asset store route as it's being deprecated, and the link to the meta UPM registry doesn't make it clear how I actually add this package or even what package to add.

That said, I don't think linking out to this page is the right solution anyway. If someone had skipped installing the examples from the initial dialog that came up, is there really no button in the Editor they can hit to bring that dialog back up? It seems crazy to me that if you skip that step, your only course of action is to go to the asset store or manually look up a UPM package bundle ID and add it to your project.

stephen-radial replied:
I've added some extra detail to the sample section, but I am a little confused, if they don't download the package from the store, or from the Meta developer website and point the package manager at a local copy (added notes about that) how would they get that package?  The Normcore download page also take the user to the Unity App store, so I would have thought this was the right place to point people on the Meta side.

maxweisel reviewed (changes requested):
I took a stab at trying to set up Meta Avatars using this guide, and I can't get past "Setting up your VR environment". I added comments at each point I got stuck, but eventually just had to give up. I felt extremely frustrated before I even got to any Normcore parts.

Will you start with a blank project, go through this guide step by step and improve the guide until it's at a level where someone someone like me can get through it without any handholding? I don't think you need to assume the reader is a total beginner, I think you can assume they are an experienced Unity / VR dev, but assume they have never used the Meta Avatars SDK and aren't familiar with any of its quirks.

stephen-radial committed:
Adding in greater depth to the Meta Developer part of the setup.

stephen-radial committed:
Adding some extra notes about pitfalls, swapping out the camera screenshot and adding one for the package name.

stephen-radial committed:
Adding note about UNITY_INSTANCING_PROCEDURAL_FUNC must be defined.

stephen-radial committed:
Removing the description of installing the Meta Avatars SDK samples, and just describe the menu item for installing our copied sample prefab.

stephen-radial committed:
Updating for feedback.  Splitting the Meta Avatar Documentation into a Meta Setup page and a Meta Avatar page.

stephen-radial commented:
Feedback from chat with Max yesterday:

- [x] Split Meta Avatar example between Meta setup and Meta Avatar
- [x] Include uploading Meta application to Meta setup document
- [x] Link Meta Setup from Meta Avatar doc
- [x] Link Normcore Setup from Meta Avatar doc
- [x] Make MetaConnect into a sample component so the user just needs to add that to their sample
- [x] Update doc to reference MetaConnect component rather than writing their own
- [x] Setup the Meta Avatar to drag/drop into scene
- [x] Change current Create Meta Avatar section into Create Meta Avatar from scratch
- [x] Move a prefabs from UPM to UPM sample to the sample dir and add a step to install samples
- [x] Add a common issues section

- [x] Look for a way to avoid the 0,0,0 issue.
  Determined this was caused because a newly added gameobject will be centered on the scene view when created rather than 0,0,0.  Set the game object to 0,0,0 as part of the "Setup Meta Avatar" button to help avoid this issue.  Seemed like the safest way, incase the user had some reason they wanted things not at 0,0,0

- [x] Try adding scoped registry to blank project to see if we can import normcore
    Yes, that seems to work, but it does default to the highest, which in the case of Normcore is the 3.0.0 alpha, which I don't think is what we would usually want by default?  If we want to install a specific version, we might need to install through an editor script.

- [x] See if there's a way to ask the user to install samples
     There isn't a way to ask the user to install UPM samples.  We could have a second UPM that contains sample files (which is what Meta is doing), but since we want to allow users to modify out samples, using the UPM sample folder makes the most sense

- [x] Blank scene template in normcore should have tracked pose driver and anything else
    Added note to docs that sample Normcore app could be used as basis and created a separate issue for updating the sample Normcore VR project

- [x] Images should be swapped to full unity editor/web pages at end

stephen-radial committed:
Swapping out pictures for a full screen ones and adding a couple of videos.

stephen-radial commented:
Adding a note in here: Completed the full window images/video pass.  That could do with a review, but other than that, text is just waiting for some wording changes from Max.

maxweisel committed:
Move the docs pages into a bespoke meta folder. Leaving the old ones around for reference.

maxweisel committed:
Remvoe redudant notes.

stephen-radial committed:
Adding sample install-meta-platform.mp4.

stephen-radial committed:
Changing resolution of install-meta-platform.mp4 to 2880x1620 so it can be cropped appropriately.

maxweisel commented on docs/guides/meta/meta-avatars.md:
```diff
@@ -0,0 +1,57 @@
+<video width="100%" controls><source src={metaPlatform} /></video> 
+
+## Set up Meta Avatars
+Import the VR examples (TODO: Are we including this in VR Examples or does the Normcore Meta package have its own samples?). Drag the "Realtime + Meta Avatar Player" into your scene.
```
@stephen-radial I'm trying the guide out now, and I realize I forgot the step where we drag in the components that are required in the scene like `OvrAvatarManager`. I'd like to do that step first thing in this section. Which also means the rest of the videos in this section need to show that prefab in the Hierarchy panel.

maxweisel replied:
Given that the only prefab that has OvrAvatarManager on it is in the Meta Avatar samples, I'd love to just bundle the prefab itself in our package (an ideally give it a new GUID so it doesn't conflict)

maxweisel replied:
It looks like their `OvrAvatarManagerBase` prefab isn't even enough smh lol. The `MetaAvatarSdkManager` + `MetaAvatarShaderManager` combo you've been copying out of their SimpleExample scene is what we should bundle as its own prefab in our package. Ideally without any child prefab references. It should reference the scripts directly.

stephen-radial committed:
Adding videos for the meta avatar page.

stephen-radial committed:
Uploading videos/images for the meta-platform-prerequisites page.

stephen-radial committed:
Adding TODO note about needing a rift app to run in editor.

stephen-radial committed:
Redoing the first 3 videos of the Meta Prerequisites doc

stephen-radial committed:
Release channels/email invite prerequisit video.

stephen-radial committed:
Meta avatar videos pass 2.

stephen-radial committed:
3rd pass on the custom avatar videos.

stephen-radial committed:
Double wave video

stephen-radial committed:
Videos with new samples

maxweisel committed:
Final copy pass on the guide.

NormalMark committed:
Change note -> danger about entitlements check

NormalMark committed:
Oculus => Meta XR in project settings

maxweisel commented on docs/guides/meta/meta-platform-prerequisites/oculus-fixes.png:
fwiw I think we should keep the Unity editor in these screenshots and it should be configured where the person following this guide would expect in terms of what scene is open and how it's set up in the background. I think the context is important when following a guide like this as a beginner.

maxweisel commented on docs/guides/meta/meta-platform-prerequisites.md:
```diff
@@ -66,13 +66,13 @@ If you're using Meta Avatars, install the [Meta Avatars SDK](https://assetstore.
 ## Initialize the Oculus platform at runtime
 
-Developers often forget to initialize the Oculus platform at runtime which can lead to a host of hard to track down bugs. You can do this in your own components by calling `Oculus.Platform.Core.Initialize()`, however the [Meta XR Platform SDK](https://assetstore.unity.com/packages/tools/integration/meta-xr-platform-sdk-262366) comes with an Entitlement Check sample. We typically recommend this ap ...
+Developers often forget to initialize the Oculus platform at runtime which can lead to a host of hard to track down bugs. You can do this in your own components by calling `Oculus.Platform.Core.Initialize()`, however the [Meta XR Platform SDK](https://assetstore.unity.com/packages/tools/integration/meta-xr-platform-sdk-262366) comes with an Entitlement Check sample that you can import using the P ...
```
nit but "import using _the_ Package manager" -> "import using Package Manager"

maxweisel committed:
Fix "the Package Manager" -> "Package Manager"

maxweisel committed:
Fix copy TODOs

NormalMark committed:
Updated AvatarSDKManager setup

NormalMark committed:
Revert "Updated AvatarSDKManager setup"

This reverts commit 89b51fc7d7f0184e8bdac5e3c86665de354ad146.

NormalMark committed:
Updated AvatarSDKManagerMeta video (fixed)

maxweisel committed:
Final copy tweaks
